### ISSUE TITLE
no-redeclare does not work with renaming in destructures inside for-loop

### ISSUE THREAD

**Issue by zachdini** [eslint/eslint#8071]
I'm running:
eslint: v3.13.1
node: v6.9.1
npm: 3.10.8
Babel-ESLint: v7.1.1

Here's the code:
```js
function foo(params) {
    for (let i = 0; i < params.length; ++i) {
        const { value: paramValue } = params[i];
        const paramValue = params[i].value;
    }
}
```

I expect to get a no-redeclare error thrown by eslint. However I am not receiving any errors. ESLint just shows that there are no errors. If I get rid of the for-loop, ESLint properly throws errors.

**Comment by not-an-aardvark** [comment]
This is actually a syntax error -- it should cause a parsing failure, since variables declared with `const` can't be redeclared.

Also see: https://github.com/ternjs/acorn/issues/487

**Comment by mysticatea** [comment]
I could not reproduce this on our online demo.

![image](https://cloud.githubusercontent.com/assets/1937871/22913058/fa36908a-f2ab-11e6-8ee3-4589a9e8c84f.png)

**Comment by mysticatea** [comment]
Also, I could not reproduce this with `babel-eslint`.

**Comment by eslintbot** [comment]
Hi @zachdini, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by zachdini** [comment]
@not-an-aardvark Not sure how I missed that. Thanks for the info. Didn't cause a parsing error so I'll have to look into that.

**Comment by alberto** [comment]
Closing this issue as it looks like the question has been answered. Please feel free to visit us in the ESLint Gitter if you have any other issues!
